test/jdk/javax/swing/KeyboardManager/bug4345798.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2000, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 4345798
+ * @summary Tests if Pressing enter to dismiss menu works when a JRootPane
+ * has a default button.
+ * @key headful
+ * @run main bug4345798
+ */
+
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.JRootPane;
+import javax.swing.SwingUtilities;
+
+public class bug4345798 {
+    private static JFrame f;
+    private static JButton b;
+    private static JMenu menu;
+    private static volatile boolean passed = true;
+    private static volatile Point p;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(() -> {
+                f = new JFrame("bug4345798");
+                JMenuBar mbar = new JMenuBar();
+                JMenuItem item = new JMenuItem("Open...");
+                menu = new JMenu("File");
+                item.addActionListener(new TestActionListener());
+                menu.add(item);
+                mbar.add(menu);
+
+                f.setJMenuBar(mbar);
+
+                b = new JButton("Default");
+                b.addActionListener(new TestActionListener());
+                f.getContentPane().add(b);
+                JRootPane rp = f.getRootPane();
+                rp.setDefaultButton(b);
+
+                f.setSize(200, 200);
+                f.setLocationRelativeTo(null);
+                f.setVisible(true);
+                b.requestFocus();
+            });
+
+            Robot robot = new Robot();
+            robot.setAutoDelay(100);
+            robot.delay(1000);
+            robot.waitForIdle();
```

Comment:
Usually, we do waitForIdle followed by delay so that all events processing is done and then wait further 1sec to let frame visible..

Reply:
Fixed, thanks